cli/types.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+type OptionTypeConstructor<T> = T extends boolean ? BooleanConstructor
+  : T extends number ? NumberConstructor
+  : T extends string ? StringConstructor
+  : BooleanConstructor | NumberConstructor | StringConstructor;
+
+export interface Option<T> {
+  name: string;
+  alias?: string;
+  description?: string;
+  default?: T;
+  type?: OptionTypeConstructor<T>;
+  negatable?: boolean;
+  value?: {
+    name: string;
+    optional?: boolean;
+    multiple?: boolean;
+    requireEquals?: boolean;
+  };
+  fn?: (value: T) => T | void;
+}
+
+export interface Argument<T = unknown> {
+  name: string;
+  description?: string;
+  multiple?: boolean;
+  optional?: boolean;
+  fn?: (value: T) => T | void;
+}
```

Comment:
2 things:

1. Do the positional arguments need a name? I guess it's useful for generating help later, but they might as well be returned as a tuple. I think it's simpler.
2. Maybe instead of `multiple` and `optional` flags on arguments, have separate lists of `requiredArguments`, `optionalArguments`, and a single `restArgument` property on `Command`. That way it can be enforced that required arguments always come first and there's at most one rest (multiple) argument at the end. It doesn't make sense to have them in any other order.

Reply:
1. That is exactly right, the name property as well as the nesting is intended for the (future) stringification of an option.
2. That gets more complicated, if one intends to extend the functionality with lets for example `standalone` options or `conflictingWith` options. So I think it is good to have these properties in the option object itself.